docs/site/Parsing-requests.md

```diff
@@ -0,0 +1,159 @@
+the controller function like below:
+
+```ts
+@put('/todos/{id}')
+async replaceTodo(
+  @param.path.number('id') id: number,
+  @requestBody() todo: Todo,
+): Promise<boolean> {
+  // NO need to do the "string to number" convertion as below
+  id = +id;
+  return await this.todoRepo.replaceById(id, todo);
+}
+```
+
+### Validation
+
+Validations are applied on the parameters and the request body data. They also
+use OpenAPI specification as the reference to infer the validation rules.
+
+#### Parameters
+
+We have the data type safety check for the parameters parsed from header, path,
+query. For example, if a parameter should be an integer, then a number with
+decimal like "1.23" will be rejected.
+
+You can specify a parameter's type by calling shortcut decorators of `@param`
+like `@param.query.integer()`. A list of available shortcuts can be found in the
+[api documents](https://apidocs.strongloop.com/@loopback%2fdocs/openapi-v3.html#param).
+And please check [parameter decorators](Decorators.md#parameter-decorator) for
+how to decorate the controller parameter.
+
+Here are our default validation rules for each type:
+
+- number: validated by `isNaN(Number(data))`.
+- integer: validated by `Number.isInteger(data)`.
+- long: validated by `Number.isSafeInteger(data)`.
+- date-time: should be a valid date-time defined in
+  [RFC3339](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14).
+- date: should be a valid full-date defined in
+  [RFC3339](https://xml2rfc.tools.ietf.org/public/rfc/html/rfc3339.html#anchor14).
+- boolean: after converted to all upper case, should be one of the following
+  values: `TRUE`, `1`, `FALSE` or `0`.
+
+#### Request Body
+
+The data from request body is validated against its OpenAPI schema
+specification. We use module [AJV](https://github.com/epoberezkin/ajv) to
+perform the validation, which validates data with a JSON schema generated from
+the OpenAPI schema specification.
+
+Take operation `replaceTodo` as an instance again:
+
+```ts
+import {Todo} from './models/todo-model.ts';
+
+...
+  @put('/todos/{id}')
+  async replaceTodo(
+    @param.path.number('id') id: number,
+    @requestBody() todo: Todo,
+  ): Promise<boolean> {
+    return await this.todoRepo.replaceById(id, todo);
+  }
+...
+```
+
+The request body specification is defined by applying `@requestBody()` to
+argument `todo`, and the schema specification inside it is inferred from its
+type `Todo`. The type is exported from a `Todo` model.
+
+_See [model](Model.md) to know more details about how to decorate a model class_
+
+When the `PUT` method `/todo/{id}` get called, the `todo` instance from request
+body will be validated with a well defined specification.
+
+Section [RequestBody Decorator](Decorators.md#requestbody-decorator) has a very
+detailed explanation of the various ways to provide a request body
+specification. It would be a redundant to repeat it here. So make sure you read
```

Comment:
I don't think we need the last two sentences

Reply:
hmm I do want to emphasize "make sure you read that section", I deleted the fist sentence.

Reply:
How about this: `Validation of model objects is heavily dependent on its OpenAPI Schema defined in/by @requestBody decorator. Please refer to the documentation on [@requestBody decorator](link) to get a comprehensive idea of defining custom validation rules for your models.`

Reply:
cool 😎 !